app/core/event_handler.py

```diff
@@ -52,6 +53,39 @@ async def handle_event(premis_event: PremisEvent) -> None:
         )
         return
 
+    # If there are multiple items for the PID, we need the one that isn't a fragment
+    if result["TotalNrOfResults"] > 1:
+        original_fragment_id = next(
+            (
+                item["Internal"]["FragmentId"]
+                for item in result["MediaDataList"]
+                if not item["Internal"]["IsFragment"]
+            )
+        )
+    elif result["TotalNrOfResults"] == 1:
+        original_fragment_id = result["MediaDataList"][0]["Internal"]["FragmentId"]
```

Comment:
Do we need this? The resultset is an array anyway, even with only one element...?